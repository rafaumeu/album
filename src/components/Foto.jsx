import PropTypes from 'prop-types'
import React from 'react'

const Foto = ({ dados, setFotoAmpliada }) => {
  return (
    <div className='foto' onClick={() => setFotoAmpliada(dados)}>
      <img src={dados.urls.small} alt={dados.alt_description} />
    </div>
  )
}

Foto.propTypes = {
  dados: PropTypes.shape({
    urls: PropTypes.shape({
      small: PropTypes.string.isRequired,
    }).isRequired,
    alt_description: PropTypes.string,
  }).isRequired,
  setFotoAmpliada: PropTypes.func.isRequired,
}

export default Foto
