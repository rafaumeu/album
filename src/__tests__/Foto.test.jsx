import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import { vi } from 'vitest'

import Foto from '../components/Foto'

describe('<Foto />', () => {
  const fotoData = {
    urls: { small: 'https://example.com/image.jpg' },
    alt_description: 'Alt Description',
  }
  it('renders without crashing', () => {
    render(<Foto dados={fotoData} />)
  })

  it('renders image with correct alt text', () => {
    const { getByAltText } = render(<Foto dados={fotoData} />)
    const imageElement = getByAltText('Alt Description')

    expect(imageElement).toBeInTheDocument()
  })
  it('calls setFotoAmpliada when clicked', () => {
    const setFotoAmpliadaMock = vi.fn()
    const { getByAltText } = render(
      <Foto dados={fotoData} setFotoAmpliada={setFotoAmpliadaMock} />
    )

    const imageElement = getByAltText('Alt Description')

    fireEvent.click(imageElement)

    expect(setFotoAmpliadaMock).toHaveBeenCalledWith(fotoData)
  })
  it('renders image with correct alt text for different data', () => {
    const fotoDataWithDifferentAlt = {
      urls: { small: 'https://example.com/another-image.jpg' },
      alt_description: 'Another Alt Description',
    }
    const { getByAltText } = render(<Foto dados={fotoDataWithDifferentAlt} />)
    const imageElement = getByAltText('Another Alt Description')

    expect(imageElement).toBeInTheDocument()
  })
  it('renders image with correct alt text for different data', () => {
    const fotoDataWithDifferentAlt = {
      urls: { small: 'https://example.com/another-image.jpg' },
      alt_description: 'Another Alt Description',
    }
    const { getByAltText } = render(<Foto dados={fotoDataWithDifferentAlt} />)
    const imageElement = getByAltText('Another Alt Description')

    expect(imageElement).toBeInTheDocument()
  })
  it('matches snapshot', () => {
    const { container } = render(
      <Foto dados={fotoData} setFotoAmpliada={() => {}} />
    )
    expect(container).toMatchSnapshot()
  })
})
