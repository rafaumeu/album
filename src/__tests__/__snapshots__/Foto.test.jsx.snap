// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Foto /> > matches snapshot 1`] = `
<div>
  <div
    class="foto"
  >
    <img
      alt="Alt Description"
      src="https://example.com/image.jpg"
    />
  </div>
</div>
`;
